### DISCUSSION TITLE
Add ability to generate dependency change table for a given tags/branches

### DISCUSSION THREAD

**Discussion by x86demon** [composer/composer#11756] [General]
I was really excited about the functionality presented at SymfonyCon2023 where release notes contain information about changed composer dependencies.
It's a common task to create release notes for projects that use composer as a package manager. It would be very useful to be able to generate such reports for a given pair of tags/branches/commits.

What do you think about this?

Good implementation example is https://github.com/JBZoo/Composer-Diff but would be nice to have something similar out of the box or as a composer plugin.

**Comment by naderman** [comment]
Hey!

I think what you are referring to is "Update Review" which is a feature of our paid product Private Packagist: https://packagist.com/features/update-review If you're interested in using Private Packagist to do this for your projects and it may need adjustments, please email us at contact@packagist.com to discuss what may be possible and what pricing plan may be appropriate for you.

Adding something to Composer to comment on pull requests is certainly out of Composer's scope. And adding a command to diff composer lock files is probably best suited to a separate tool or a plugin, because it seems like a bit of a niche problem, that people have various different other solutions for and different needs/ideas on what exactly it should be doing.

Best,
Nils
